README.md

```diff
@@ -1,2 +1,233 @@
+decentralised applications. We want a “roadmap without potholes” for providing
+stronger privacy assurances than Tor for both senders and receivers, while also
+scaling well and providing short-term message storage for offline users. In
+other words, we accept that a project this large requires a piecemeal approach
+but we shall understand and avoid design dead ends that prevent either scaling
+or rigorous analysis of anonymity properties.
+
+We think mix networks occupy a sweet spot in which nodes run extremely efficiently but rigorous analysis remains possible, if challenging.  There are designs like Tor with more efficient designs, but they cannot provide rigorous anonymity properties.  There are also numerous academic schemes designed to support rigorous analysis, but at extreme sacrifices in efficiency.  We choose a Sphinx-like packet format because it’s efficient, adaptable enough, and has good security proofs.  
+
+We leave actually analyzing our scheme for future work in collaboration with academics.  We shall however use Poison mixing and cover traffic strategy similar to Loopix, which appears analyzable although academic work to do so remains ongoing.  We concede that mix networks impose latency on users, but any faster design would definitely not admit rigorous analysis of anonymity properties, and could not credibly claim to be stronger than Tor.  
+
+We shall tweak Sphinx slightly to accommodate both receiver anonymity and
+short-term message storage simultaneously, which may require updating its
+security proofs eventually. We consider short-term message storage essential to
+user experience and advise against doing it via a second layer protocol.
+
+We foresee the public key infrastructure (PKI) being the ultimate scaling
+bottleneck for all existing anonymity scheme designs. We could avoid this with
+gossip protocols but these enable epistemic attacks. We largely leave this to
+future work, but suggest investigating a verifiable gossip based protocol. We
+shall therefore use an insecure gossip based protocol initially with the hope
+that it meshes best with later designs. If we used a more secure design inspired
+by the Tor consensus then we might assumptions elsewhere that limit scalability
+later.
+
+We know rewards for node operators remains a contentious question in the
+anonymity community with seemingly unforeseeable consequences. We nevertheless
+think rewards represents our best hope for a network large enough to challenge
+today’s centralized providers that operate on surveillance capitalism. We do not
+imagine rewards obliviate the need to steward relay operator culture, possibly
+quite the opposite. Messaging types
+We’re focusing on one-to-one messaging for now.  We actually do require messaging layer crypto, even after all the mix net layers, so expect an Axolotl-like ratchet for this.  
+
+We can adapt our short-term message storage plans for small group messaging, but
```

Comment:
Ditto above, to me group chat seems like a concern for higher layers. See SoK Table 2 http://cacr.uwaterloo.ca/techreports/2015/cacr2015-02.pdf and Briar's protocol stack https://code.briarproject.org/briar/briar/wikis/A-Quick-Overview-of-the-Protocol-Stack for motivation

Reply:
I believe a native group chat and pubsub schemes would dramatically improve performance latency for applications, but these come in some "SURB usage" or "client to node relationship" layer above the packet format and route builder aka PKI, but below the "conversational security" layer.  It's possible this layer has a 1-to-1 relationship with modules in the "conversational security" layer, and dApp developers see them together, but they still feel like different layers from an implementation point of view.